haystack/components/preprocessors/document_cleaner.py

```diff
@@ -42,6 +57,7 @@ def __init__(
         remove_repeated_substrings: bool = False,
         remove_substrings: Optional[List[str]] = None,
         remove_regex: Optional[str] = None,
+        id_generator: Callable[[Document, Document], str] = DEFAULT_ID_GENERATOR,
```

Comment:
One last nitpick. Please make this parameter Optional and assign it to None. 

In the body of init do the following:
```
self.id_generator = id_generator or DEFAULT_ID_GENERATOR
```
Although this is not strictly related to mutable default Python gotcha it is better for consistency as this is what we are trying to keep/maintain across all classes. Thanks for understanding.

Reply:
Done